pkgs/os-specific/linux/ch9344/default.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv, lib, fetchzip, kernel }:
+
+stdenv.mkDerivation rec {
+  pname = "ch9344";
+  version = "1.9";
+
+  src = fetchzip {
+    name = "CH9344SER_LINUX.zip";
+    url = "https://www.wch.cn/downloads/file/386.html#CH9344SER_LINUX.zip";
+    hash = "sha256-g55ftAfjKKlUFzGhI1a/O7Eqbz6rkGf1vWuEJjBZxBE=";
+  };
+
+  patches = lib.optionals (lib.versionAtLeast kernel.modDirVersion "6.1") [
+    # https://github.com/torvalds/linux/commit/a8c11c1520347be74b02312d10ef686b01b525f1
+    ./fix-incompatible-pointer-types.patch
+  ];
+
+  sourceRoot = "${src.name}/driver";
+  hardeningDisable = [ "pic" ];
```

Comment:
Would be good to leave a comment saying why this was needed

Reply:
I took that line from [here](https://nixos.wiki/wiki/Linux_kernel#Developing_out-of-tree_kernel_modules), which says:
```
For kernel modules it is necessary to disable pic in compiler hardenings as the kernel need different compiler flags. 
```

So it sounds to me that this is required and expected when creating a package for out-of-tree modules. For reference I use `r8125` package as the starting point and that package contains the same line without any comment. Same for a few other kernel modules (exfat, rtl8*) that I just checked right now.

Please let me know if you still think some explanation is needed here.

Reply:
I took that line from [here](https://nixos.wiki/wiki/Linux_kernel#Developing_out-of-tree_kernel_modules), which says:
```
For kernel modules it is necessary to disable pic in compiler hardenings as the kernel need different compiler flags. 
```

So it sounds to me that this is required and expected when creating a package for out-of-tree modules. For reference I use `r8125` package as the starting point and that package contains the same line without any comment. Same for a few other kernel modules (exfat, rtl8*) that I just checked right now.

Please let me know if you still think some explanation is needed here.

Reply:
Ok, it's fine